development/gcbuild/main.go

```diff
@@ -0,0 +1,340 @@
+		tag = "PR-" + pr
+	} else {
+		// build a tag from commit SHA
+		t, err := tags.NewTag(
+			tags.CommitSHA(sha))
+		if err != nil {
+			return fmt.Errorf("could not create tag: %w", err)
+		}
+
+		tagger := tags.Tagger{TagTemplate: `v{{ .Date }}-{{ .ShortSHA }}`}
+		tag, err = tagger.BuildTag(t)
+		if err != nil {
+			return fmt.Errorf("could not build tag: %w", err)
+		}
+	}
+
+	// TODO (@Ressetkk): custom staging bucket implementation and re-using source code in builds
+	//if o.stagingBucket != "" {
+	//
+	//}
+
+	vs, err := getVariants(o)
+	if err != nil {
+		return err
+	}
+
+	if len(vs) == 0 {
+		// variants.yaml file not present or either empty. Run single build.
+		// subs act as overrides here for YAML substitutions
+		subs := make(map[string]string)
+
+		err = run(o, "build", repo, tag, subs)
+		if err != nil {
+			return fmt.Errorf("build encountered error: %w", err)
+		}
+		img := getImageNames(repo, tag, "", config.Images)
+		fmt.Println("Successfully built image:", img)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(vs))
+	var errs []error
+	for k, v := range vs {
+		go func(variant string, env map[string]string) {
+			defer wg.Done()
+			env["_VARIANT"] = variant
+			if err := run(o, variant, repo, tag, env); err != nil {
+				errs = append(errs, fmt.Errorf("job %s ended with error: %w", variant, err))
+				fmt.Printf("Job %s ended with error: %s.\n", variant, err)
+			} else {
+				img := getImageNames(repo, tag, variant, config.Images)
+				fmt.Println("Successfully build image:", img)
+				fmt.Printf("Job %s finished successfully.\n", variant)
+			}
+		}(k, v)
+	}
+	wg.Wait()
+	return errutil.NewAggregate(errs)
+}
+
+// validateOptions handles flag validation. All checks should be provided here
+func validateOptions(o options) error {
+	var errs []error
+	if o.project == "" {
+		errs = append(errs, fmt.Errorf("--project flag is missing"))
+	}
+	if o.variant != "" && o.variantsFile == "" {
+		errs = append(errs, fmt.Errorf("variant flag is defined, but variants.yaml flag is missing"))
+	}
+	return errutil.NewAggregate(errs)
+}
+
+// checkDependencies checks if required binaries are present in PATH
+// If they are not present, then returns proper error message.
+func checkDependencies() error {
+	deps := []string{
+		"gcloud",
+		"gsutil",
+		"git",
+	}
+	var errs []error
+	for _, c := range deps {
+		if _, err := exec.LookPath(c); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errutil.NewAggregate(errs)
+}
+
+func (o *options) gatherOptions(fs *flag.FlagSet) *flag.FlagSet {
+	fs.BoolVar(&o.silent, "silent", false, "Do not push build logs to stdout")
+	fs.StringVar(&o.buildDir, "build-dir", ".", "Path to build directory")
+	fs.StringVar(&o.configFile, "config-file", "cloudbuild.yaml", "Path to cloudbuild.yaml file relative to build-dir")
+	fs.StringVar(&o.variantsFile, "variants-file", "", "Name of variants file relative to build-dir")
+	fs.StringVar(&o.variant, "variant", "", "Define which variant should be built")
```

Comment:
It's added if someone wants to use this tool as standalone application. They can define only one variant to be built, instead of building all variants. I don't think we will be using this in the pipelines.